vast/opensetAlgos/WPL.py

```diff
@@ -0,0 +1,140 @@
+
+"""
+Author: Akshay Raj Dhamija
+
+Weibull Prototype Learning (WPL)
+
+"""
+
+import torch
+import itertools
+from ..tools import pairwisedistances
+from ..DistributionModels import weibull
+from typing import Iterator, Tuple, List, Dict, OrderedDict
+
+
+def WPL_Params(parser):
+    WPL_params = parser.add_argument_group("WPL params")
+    WPL_params.add_argument(
+        "--tailsize",
+        nargs="+",
+        type=float,
+        default=[1.0],
+        help="tail size to use default: %(default)s",
+    )
+    WPL_params.add_argument(
+        "--distance_multiplier",
+        nargs="+",
+        type=float,
+        default=[1.0],
+        help="distance multiplier to use default: %(default)s",
+    )
+    WPL_params.add_argument(
+        "--default_scale",
+        nargs="+",
+        type=float,
+        default=[1.0],
+        help="Weibull scale to use when the number of uniqe element is less than 5 default: %(default)s",
+    )
+    WPL_params.add_argument(
+        "--default_shape",
+        nargs="+",
+        type=float,
+        default=[0.1],
+        help="Weibull shape to use when the number of uniqe element is less than 5 default:  %(default)s",
+    )
```

Comment:
**Algo Breakdown:** Please try to simplify or break down your algorithm. Your WPL algo has two parts one is per dimension weibull and the other is how you address the case where weibull is needed for less than 5 samples/values. I am asking to separate this part from your core algorithm.
If tomorrow someone comes up with a new way to estimate `default_scale` and `default_shape` rather than fixing default values (let's say a mean from other class values, which can't be found until the algorithm has been run on the entire dataset) would you create a new `WPL.py` and call it a new algorithm even though the per dimension weibull idea is the same?

**Reducing Computation:** `default_scale` and `default_shape` are not similar to `tail_size, cover_threshold, and distance_multiplier` of EVM. These EVM parameters were being used as a list to reduce redundant computation. In your case, you will be recalling the `mr.FitHigh` function using the same parameters when your `default_scale` and `default_shape` change, even though they do not impact the result in this case. That is you are adding computational overhead rather than reducing it.
`default_scale` and `default_shape` are simply being used to replace scale and shape values where none could be found, not reduce computation.

Reply:
Both of the comments `Algo Breakdown` and `Reducing Computation` are correct. @scruz13 please make required changes.